### ISSUE TITLE
Update the 'test/error.js' to test with mutate-fs vesion 2.1.0

### ISSUE THREAD

**Issue by akhvar** [isaacs/ignore-walk#1]
I am trying to package ignore-walk module as a Debian node-package. I can't proceed because the test fails on 'test/errors.js' with mutate-fs version 2.1.0 (already packaged) where as it works with mutate-fs version 1.1.0. The error output is added below.
```
# Subtest: test/errors.js
    # Subtest: error only emits once
        # Subtest: async
            ok 1 - should be equal
            1..1
        ok 1 - async # time=18.265ms
        
        # Subtest: sync
            not ok 1 - expected to throw: Error poop
              ---
              wanted:
                message: poop
                name: Error
                callstack: |-
                  Error: trace
                      at Object.fs.(anonymous function) [as lstat] (/usr/lib/nodejs/mutate-fs/index.js:103:23)
                      at Walker.stat (/<<PKGBUILDDIR>>/index.js:161:39)
                      at filtered.forEach.filt (/<<PKGBUILDDIR>>/index.js:139:14)
                      at Array.forEach (native)
                      at Walker.filterEntries (/<<PKGBUILDDIR>>/index.js:135:16)
                      at Walker.onReaddir (/<<PKGBUILDDIR>>/index.js:69:14)
                      at fs.readdir (/<<PKGBUILDDIR>>/index.js:46:42)
                      at FSReqWrap.oncomplete (fs.js:123:15)
              found:
                stack: |-
                  Error: poop
                      at Test.t.test.t (/<<PKGBUILDDIR>>/test/errors.js:13:16)
                      at bound (domain.js:280:14)
                      at Test.runBound (domain.js:293:12)
                      at Test.main (/usr/lib/nodejs/tap/lib/test.js:218:20)
                      at TAP.Test.writeSubComment (/usr/lib/nodejs/tap/lib/test.js:353:3)
                      at loop (/usr/lib/nodejs/function-loop/index.js:35:15)
                      at TAP.Test.runBeforeEach (/usr/lib/nodejs/tap/lib/test.js:677:5)
                      at TAP.Test.processSubtest (/usr/lib/nodejs/tap/lib/test.js:317:10)
                      at TAP.Test.process (/usr/lib/nodejs/tap/lib/test.js:261:12)
                      at TAP.Test.sub (/usr/lib/nodejs/tap/lib/test.js:158:8)
                message: poop
                callstack: |-
                  Error: trace
                      at Object.fs.(anonymous function)._ [as fstatSync] (/usr/lib/nodejs/mutate-fs/index.js:111:23)
                      at tryStatSync (fs.js:464:13)
                      at Object.fs.readFileSync (fs.js:512:12)
                      at cleanTapYamlObject (/usr/lib/nodejs/tap/lib/clean-yaml-object.js:27:32)
                      at objToYaml (/usr/lib/nodejs/tap/lib/obj-to-yaml.js:7:9)
                      at diags (/usr/lib/nodejs/tap/lib/diags.js:6:11)
                      at tpMessage (/usr/lib/nodejs/tap/lib/point.js:37:37)
                      at new TestPoint (/usr/lib/nodejs/tap/lib/point.js:16:18)
                      at Test.pR [as printResult] (/usr/lib/nodejs/tap/lib/test.js:542:16)
                      at Test.fail (/usr/lib/nodejs/tap/lib/test.js:465:8)
                name: Error
              pattern:
                message: poop
                name: Error
                callstack: |-
                  Error: trace
                      at Object.fs.(anonymous function) [as lstat] (/usr/lib/nodejs/mutate-fs/index.js:103:23)
                      at Walker.stat (/<<PKGBUILDDIR>>/index.js:161:39)
                      at filtered.forEach.filt (/<<PKGBUILDDIR>>/index.js:139:14)
                      at Array.forEach (native)
                      at Walker.filterEntries (/<<PKGBUILDDIR>>/index.js:135:16)
                      at Walker.onReaddir (/<<PKGBUILDDIR>>/index.js:69:14)
                      at fs.readdir (/<<PKGBUILDDIR>>/index.js:46:42)
                      at FSReqWrap.oncomplete (fs.js:123:15)
              at:
                line: 31
                column: 7
                file: test/errors.js
                type: Test
                function: t.test.t
              stack: |
                Test.t.test.t (test/errors.js:31:7)
                Walker.w.on.er (test/errors.js:26:9)
                Walker.emit (index.js:39:21)
                fs.(anonymous function) (index.js:163:14)
                Timeout._ [as _onTimeout] (/usr/lib/nodejs/mutate-fs/index.js:106:7)
              ...
            
            1..1
            # failed 1 test
        not ok 2 - sync # time=41.828ms
        
        1..2
        # failed 1 of 2 tests
    not ok 1 - error only emits once # time=72.258ms
    
    # Subtest: readdir error
        # Subtest: async
            ok 1 - should be equal
            1..1
        ok 1 - async # time=4.128ms
        
        # Subtest: sync
            not ok 1 - expected to throw: Error poop
              ---
              wanted:
                message: poop
                name: Error
                callstack: |-
                  Error: trace
                      at Object.fs.(anonymous function) [as readdir] (/usr/lib/nodejs/mutate-fs/index.js:103:23)
                      at Walker.start (/<<PKGBUILDDIR>>/index.js:45:8)
                      at Test.t.test.t (/<<PKGBUILDDIR>>/test/errors.js:47:8)
                      at bound (domain.js:280:14)
                      at Test.runBound (domain.js:293:12)
                      at Test.main (/usr/lib/nodejs/tap/lib/test.js:218:20)
                      at Test.writeSubComment (/usr/lib/nodejs/tap/lib/test.js:353:3)
                      at loop (/usr/lib/nodejs/function-loop/index.js:35:15)
                      at Test.<anonymous> (/usr/lib/nodejs/tap/lib/test.js:674:7)
                      at loop (/usr/lib/nodejs/function-loop/index.js:35:15)
              found:
                stack: |-
                  Error: poop
                      at Test.t.test.t (/<<PKGBUILDDIR>>/test/errors.js:39:16)
                      at bound (domain.js:280:14)
                      at Test.runBound (domain.js:293:12)
                      at Test.main (/usr/lib/nodejs/tap/lib/test.js:218:20)
                      at TAP.Test.writeSubComment (/usr/lib/nodejs/tap/lib/test.js:353:3)
                      at loop (/usr/lib/nodejs/function-loop/index.js:35:15)
                      at TAP.Test.runBeforeEach (/usr/lib/nodejs/tap/lib/test.js:677:5)
                      at TAP.Test.processSubtest (/usr/lib/nodejs/tap/lib/test.js:317:10)
                      at TAP.Test.process (/usr/lib/nodejs/tap/lib/test.js:261:12)
                      at TAP.pR [as printResult] (/usr/lib/nodejs/tap/lib/test.js:556:8)
                message: poop
                callstack: |-
                  Error: trace
                      at Object.fs.(anonymous function)._ [as readdirSync] (/usr/lib/nodejs/mutate-fs/index.js:111:23)
                      at WalkerSync.start (/<<PKGBUILDDIR>>/index.js:230:23)
                      at t.throws._ (/<<PKGBUILDDIR>>/test/errors.js:51:36)
                      at Test.<anonymous> (/usr/lib/nodejs/tap/lib/asserts.js:237:7)
                      at Test.ASSERT [as currentAssert] (/usr/lib/nodejs/tap/lib/test.js:444:15)
                      at Test.t.test.t (/<<PKGBUILDDIR>>/test/errors.js:51:7)
                      at bound (domain.js:280:14)
                      at Test.runBound (domain.js:293:12)
                      at Test.main (/usr/lib/nodejs/tap/lib/test.js:218:20)
                      at Test.writeSubComment (/usr/lib/nodejs/tap/lib/test.js:353:3)
                name: Error
              pattern:
                message: poop
                name: Error
                callstack: |-
                  Error: trace
                      at Object.fs.(anonymous function) [as readdir] (/usr/lib/nodejs/mutate-fs/index.js:103:23)
                      at Walker.start (/<<PKGBUILDDIR>>/index.js:45:8)
                      at Test.t.test.t (/<<PKGBUILDDIR>>/test/errors.js:47:8)
                      at bound (domain.js:280:14)
                      at Test.runBound (domain.js:293:12)
                      at Test.main (/usr/lib/nodejs/tap/lib/test.js:218:20)
                      at Test.writeSubComment (/usr/lib/nodejs/tap/lib/test.js:353:3)
                      at loop (/usr/lib/nodejs/function-loop/index.js:35:15)
                      at Test.<anonymous> (/usr/lib/nodejs/tap/lib/test.js:674:7)
                      at loop (/usr/lib/nodejs/function-loop/index.js:35:15)
              at:
                line: 51
                column: 7
                file: test/errors.js
                type: Test
                function: t.test.t
              stack: |
                Test.t.test.t (test/errors.js:51:7)
                Walker.on.er (test/errors.js:46:9)
                Walker.emit (index.js:39:21)
                fs.readdir (index.js:46:17)
                Timeout._ [as _onTimeout] (/usr/lib/nodejs/mutate-fs/index.js:106:7)
              source: |
                t.throws(_ => new WalkerSync().start(), poop)
              ...
            
            1..1
            # failed 1 test
        not ok 2 - sync # time=33.102ms
        
        1..2
        # failed 1 of 2 tests
    not ok 2 - readdir error # time=44.166ms
    
    # Subtest: readFile error
        # Subtest: async
            ok 1 - should be equal
            1..1
        ok 1 - async # time=9.911ms
        
        # Subtest: sync
            not ok 1 - expected to throw: Error poop
              ---
              wanted:
                message: poop
                name: Error
                callstack: |-
                  Error: trace
                      at Object.fs.(anonymous function) [as readFile] (/usr/lib/nodejs/mutate-fs/index.js:103:23)
                      at Walker.addIgnoreFile (/<<PKGBUILDDIR>>/index.js:88:8)
                      at newIg.forEach.e (/<<PKGBUILDDIR>>/index.js:83:29)
                      at Array.forEach (native)
                      at Walker.addIgnoreFiles (/<<PKGBUILDDIR>>/index.js:83:11)
                      at Walker.onReaddir (/<<PKGBUILDDIR>>/index.js:67:14)
                      at fs.readdir (/<<PKGBUILDDIR>>/index.js:46:42)
                      at FSReqWrap.oncomplete (fs.js:123:15)
              found:
                stack: |-
                  Error: poop
                      at Test.t.test.t (/<<PKGBUILDDIR>>/test/errors.js:59:16)
                      at bound (domain.js:280:14)
                      at Test.runBound (domain.js:293:12)
                      at Test.main (/usr/lib/nodejs/tap/lib/test.js:218:20)
                      at TAP.Test.writeSubComment (/usr/lib/nodejs/tap/lib/test.js:353:3)
                      at loop (/usr/lib/nodejs/function-loop/index.js:35:15)
                      at TAP.Test.runBeforeEach (/usr/lib/nodejs/tap/lib/test.js:677:5)
                      at TAP.Test.processSubtest (/usr/lib/nodejs/tap/lib/test.js:317:10)
                      at TAP.Test.process (/usr/lib/nodejs/tap/lib/test.js:261:12)
                      at TAP.pR [as printResult] (/usr/lib/nodejs/tap/lib/test.js:556:8)
                message: poop
                callstack: |-
                  Error: trace
                      at Object.fs.(anonymous function)._ [as readFileSync] (/usr/lib/nodejs/mutate-fs/index.js:111:23)
                      at cleanTapYamlObject (/usr/lib/nodejs/tap/lib/clean-yaml-object.js:27:32)
                      at objToYaml (/usr/lib/nodejs/tap/lib/obj-to-yaml.js:7:9)
                      at diags (/usr/lib/nodejs/tap/lib/diags.js:6:11)
                      at tpMessage (/usr/lib/nodejs/tap/lib/point.js:37:37)
                      at new TestPoint (/usr/lib/nodejs/tap/lib/point.js:16:18)
                      at Test.pR [as printResult] (/usr/lib/nodejs/tap/lib/test.js:542:16)
                      at Test.fail (/usr/lib/nodejs/tap/lib/test.js:465:8)
                      at Test.<anonymous> (/usr/lib/nodejs/tap/lib/asserts.js:28:17)
                      at Test.ASSERT (/usr/lib/nodejs/tap/lib/test.js:444:15)
                name: Error
              pattern:
                message: poop
                name: Error
                callstack: |-
                  Error: trace
                      at Object.fs.(anonymous function) [as readFile] (/usr/lib/nodejs/mutate-fs/index.js:103:23)
                      at Walker.addIgnoreFile (/<<PKGBUILDDIR>>/index.js:88:8)
                      at newIg.forEach.e (/<<PKGBUILDDIR>>/index.js:83:29)
                      at Array.forEach (native)
                      at Walker.addIgnoreFiles (/<<PKGBUILDDIR>>/index.js:83:11)
                      at Walker.onReaddir (/<<PKGBUILDDIR>>/index.js:67:14)
                      at fs.readdir (/<<PKGBUILDDIR>>/index.js:46:42)
                      at FSReqWrap.oncomplete (fs.js:123:15)
              at:
                line: 70
                column: 7
                file: test/errors.js
                type: Test
                function: t.test.t
              stack: |
                Test.t.test.t (test/errors.js:70:7)
                Walker.on.er (test/errors.js:65:9)
                Walker.emit (index.js:39:21)
                Walker.emit (index.js:37:27)
                Walker.emit (index.js:37:27)
                fs.readFile (index.js:89:17)
                Timeout._ [as _onTimeout] (/usr/lib/nodejs/mutate-fs/index.js:106:7)
              ...
            
            1..1
            # failed 1 test
        not ok 2 - sync # time=19.176ms
        
        1..2
        # failed 1 of 2 tests
    not ok 3 - readFile error # time=34.708ms
    
    1..3
    # failed 3 of 3 tests
    # time=158.248ms
not ok 5 - test/errors.js # time=312.171ms
  ---
```
Please fix test/errors.js to work with mutate-fs version 2.1.0.

**Comment by harish2704** [comment]
This issue doesn't exists with latest `mutate-fs` ( mutate-fs 2.1.1 ) version.
So, I think issue was with mutate-fs module and author fixed it in the latest version.

**Comment by akhvar** [comment]
This issue was with mutate-fs version 2.1.0. It was fixed by updating to 2.1.1
